crates/polars-ops/src/series/ops/round.rs

```diff
@@ -37,6 +37,55 @@ pub trait RoundSeries: SeriesSealed {
         polars_bail!(opq = round, s.dtype());
     }
 
+    fn round_sf(&self, significant_figures: u32) -> PolarsResult<Series> {
+        let s = self.as_series();
+        let significant_figures = if significant_figures == 0 {
+            1
+        } else {
+            significant_figures
+        };
+
+        if let Ok(ca) = s.f64() {
+            let s = ca
+                .zip_with(
+                    &ca.equal(0),
+                    &ca.apply_values(|val| round_sf(val, significant_figures)),
+                )?
+                .into_series();
+            return Ok(s);
+        }
+        // Note we do the computation on f64 floats to not lose precision
+        // when the computation is done we cast back
+        if let Ok(ca) = s.f32() {
+            let s = ca
+                .zip_with(
+                    &ca.equal(0),
+                    &ca.apply_values(|val| round_sf(val as f64, significant_figures) as f32),
+                )?
+                .into_series();
+            return Ok(s);
+        }
+        if let Ok(ca) = s.i32() {
```

Comment:
Is this the right way to allow for it working with integers?

Reply:
Could you use the `with_match_physical_numeric_polars_type` macro to cover all types?